examples/tasks/async_compute.rs

```diff
@@ -0,0 +1,114 @@
+use bevy::{prelude::*, tasks::{AsyncComputeTaskPool, Task}};
+use std::time::{Instant, Duration};
+use rand::Rng;
+use futures_lite::future;
+
+/// This example shows how to use the ECS and the AsyncComputeTaskPool
+/// to spawn, poll, and complete tasks across systems and system ticks.
+
+// Number of cubes to spawn across the x, y, and z axis
+const NUM_CUBES: u32 = 6;
+
+/// This system generates tasks simulating computationally intensive
+/// work that potentially spans multiple frames/ticks. A separate
+/// system, handle_tasks, will poll the spawned tasks on subsequent
+/// frames/ticks, and use the results to spawn cubes
+fn spawn_tasks(
+    mut commands: Commands,
+    thread_pool: Res<AsyncComputeTaskPool>,
+) {
+    for x in 0 ..NUM_CUBES {
+        for y in 0 ..NUM_CUBES {
+            for z in 0 ..NUM_CUBES {
+
+                // Spawn new task on the AsyncComputeTaskPool
+                let task = thread_pool.spawn(async move {
+
+                    let mut rng = rand::thread_rng();
+                    let start_time = Instant::now();
+                    let duration = Duration::from_secs_f32(rng.gen_range(0.05..0.2));
+                    while Instant::now() - start_time < duration {

```

Comment:
We should use `futures_lite::pending!()` to yield back in tight loops, perhaps with a separate counter so we yield every N iterations.

Reply:
I was following the examples in the ``bevy_task`` crate: 

https://github.com/bevyengine/bevy/blob/97d8e4e1793ede3df8c77ed44736e800b38ff7a4/crates/bevy_tasks/examples/busy_behavior.rs#L18

Reply:
Hmm, then we can probably leave this as-is for now and update both in a follow-up PR